### PR TITLE
Add Promise return type doc in Async client

### PR DESCRIPTION
**Pull request by sagikazarmark**

### CODE REVIEW

File: src/HttpAsyncClient.php [hunk 1]
```diff
@@ -19,7 +19,7 @@
      *
      * @param RequestInterface $request
      *
-     * @return Promise
+     * @return Promise MUST return with a PSR-7 Response.
```
Comment (joelwurtz) [inline src/HttpAsyncClient.php]: Not a fan :p

More like : 

The promise resolve a ResponseInterface or fail with a Http\Client\Exception

Comment (sagikazarmark) [inline src/HttpAsyncClient.php]: Sounds good to me.

**Comment by sagikazarmark** [comment]
Looks better?

**Comment by joelwurtz** [comment]
Would remove "MUST" as the message is more for the user, and people implementing the library must conform with the view of the user

**Comment by joelwurtz** [comment]
A part from that :+1:

**Pull request merged**